File: BI-TJV/semestral/rest-server/src/main/java/wrzecond/entity/TjvFood.java
package wrzecond.entity;

import lombok.*;

import javax.persistence.*;
import java.util.List;

@Entity
@EqualsAndHashCode(callSuper = true)
@AllArgsConstructor
@NoArgsConstructor
public class TjvFood extends TjvEntity {

    @Column(nullable = false)
    @Getter @Setter private String name;

    @Column(nullable = false)
    @Getter @Setter private int price;

    @Column(nullable = false)
    @Getter @Setter private boolean cooked;

    @ManyToMany
    @JoinTable(name = "TJV_FOOD_ALLERGEN", joinColumns = @JoinColumn(name = "food_id"),
            inverseJoinColumns = @JoinColumn(name = "allergen_id"))
    @Getter @Setter private List<TjvAllergen> allergens;

}

File: BI-KOT/semestral/rest-server/src/main/kotlin/wrzecond/entity/TjvFood.kt
package wrzecond.entity

import javax.persistence.*

@Entity
data class TjvFood (
    @Column(nullable = false) val name: String,
    @Column(nullable = false) val price: Int,
    @Column(nullable = false) val cooked: Boolean,

    @ManyToMany
    @JoinTable (
        name = "TJV_FOOD_ALLERGEN",
        joinColumns = [JoinColumn(name = "food_id")],
        inverseJoinColumns = [JoinColumn(name = "allergen_id")]
    )
    val allergens: List<TjvAllergen>,
    override val id: Int = 0
) : TjvEntity()
